### PR TITLE
Allow auth response exclusion fields to be specified externally

### PR DESCRIPTION
**Pull request by tomjackman**
**Motivation**

Based on [this feedback](https://github.com/feedhenry-raincatcher/raincatcher-user/pull/11#discussion_r90840005), added support to specify the user auth exclusion list outside of raincatcher-user. This config is defined in the same place and manner as the `securableEndpoints` config.

### CODE REVIEW

File: application.js [hunk 1]
```diff
@@ -26,7 +26,10 @@ app.use(mbaasExpress.fhmiddleware());
 
 app.use('/hello', require('./lib/hello.js')());
 app.use('/api', bodyParser.json({limit: '10mb'}));
-require('fh-wfm-user/lib/router/mbaas')(mediator, app);
+
+// list the user fields which you don't want appearing in the authentication response.
+var authResponseExclusionList = ['password'];
+require('fh-wfm-user/lib/router/mbaas')(mediator, app, authResponseExclusionList);
```
Comment (nialldonnellyfh) [inline application.js]: This is not being applied to the raincatcher-user module https://github.com/feedhenry-raincatcher/raincatcher-user/pull/11/files#diff-c4ad2ae31649a3a42bbb5013da8dec80R107

File: README.md [hunk 2]
```diff
@@ -15,3 +15,10 @@ This repository should be used in conjonction with these following repos :
 ## Starting (locally)
 
 `grunt`
+
+
+## Configuration
+
+You can choose which fields to remove from the user authentication response by specifying the field names in the `authResponseExclusionList` array in `application.js`.
+
+For Example, setting the array as `['password', 'banner']` will prevent these fields from appearing in the authentication response. By default, the `password` field is removed from the response. To allow all fields to be sent, set `authResponseExclusionList` as an empty array.
```
Comment (nialldonnellyfh) [inline README.md]: This should be in the raincatcher-user module README. It should be noted in the comment above the place where it is being used.

**Review by nialldonnellyfh — COMMENTED** [review]
Looks good. Just needs a unit test.

File: test/unit/test_application.js [hunk 3]
```diff
@@ -72,6 +72,33 @@ var runApp = function() {
   });
 };
 
+exports['test instantiate the user module with the required parameters'] = function(done) {
```
Comment (nialldonnellyfh) [inline test/unit/test_application.js]: Can you switch to the describe API for defining your tests. That's the convention we use for mocha.

See https://github.com/feedhenry-raincatcher/raincatcher-user/pull/19/files#diff-45a64b8b3cef7d7a7b79beb5d94db350R10 as an example

**Review by nialldonnellyfh — APPROVED** [review]
Small comment. Otherwise looks good.

**Comment by paolobueno** [comment]
incorporated by #29

**Pull request closed without merging**